pkg/request/batchstore.go

```diff
@@ -206,31 +186,33 @@ func (bs *BatchStore) Fetch(ctx context.Context) []interface{} {
 		}
 	}()
 
-	for {
```

Comment:
I think I remember now why I had a for here. When you use a condition variable you often accommodate multiple waiters. Since we only have a single waiting goroutine, we don't need the for loop, as there is only one goroutine that can be woken up.